bloom-inference-scripts/bloom-ds-inference.py

```diff
@@ -100,7 +100,7 @@ def get_checkpoint_files(model_name_or_path):
 
 
 model_name = args.name
-infer_dtype = args.dtype
+infer_dtype = args.dtype if args.dtype != 'int4' else 'int8'
```

Comment:
would it make for a more user-friendly API to 

1. keep the `dtype`  intact
2. drop ` quantization_bits`
3. let `deepspeed.init_inference` derive the number of bit from `dtype`?

not only the currently suggested override is confusing, I fail to see what purpose serves carrying the same information in `dtype` and and `quantization_bits`.

Reply:
oh, wait, `torch.init4` still doesn't exist, does it?

let's find the feature request.

Reply:
still not implemented https://github.com/pytorch/pytorch/issues/74627

so that's why you had to do the odd workarounds, right?

Reply:
I guess we can drop it once its implemented @stas00 ?
For now, this might be the best way to do it.

Reply:
see https://github.com/huggingface/transformers-bloom-inference/pull/37#issuecomment-1320686912

Reply:
it's pointless to wait, since they won't have `int3` and `int12`

Reply:
> would it make for a more user-friendly API to
> 
> 1. keep the `dtype`  intact
> 2. drop ` quantization_bits`
> 3. let `deepspeed.init_inference` derive the number of bits from `dtype`?
> 
> not only the currently suggested override is confusing, I fail to see what purpose serves carrying the same information in `dtype` and and `quantization_bits` twice

@stas00 and @RezaYazdaniAminabadi - just clarifying that we have introduced a new DeepSpeedInferenceConfig that can be passed to init_inference. We are keeping it backwards compatible but if we are okay to make changes to this file, I would advocate for writing a config dictionary for DeepSpeed and pass that to init_inference instead of the various kwargs. Please see here for an example: https://gist.github.com/awan-10/6e3d5c756be3a876522e860c6bbf702d#file-bloom-ds-inference-py-L173 

Also, see the docs for the new config: https://deepspeed.readthedocs.io/en/latest/inference-init.html

Reply:
That definitely works.

@awan-10, may I suggest you make the inference `config` accept `dict_or_path` just like `zero` does? it might be for some users easier to write out a separate file.

Reply:
@stas00 - thanks for the suggestion. Created an issue so we can track it: https://github.com/microsoft/DeepSpeed/issues/2532. Mike and I will work on it.

Reply:
Thank you very much, @awan-10